<a href="https://colab.research.google.com/github/JALVARADORUIZ/DataAnalytic-PJUD/blob/master/tarea1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
# Importing the libraries
import pandas as pd
import numpy as np

Carga de Datos y visualización general

In [4]:
# Carga del fichero desde el enlace web y creación del dataframe
url = 'https://github.com/md-lorente/data/raw/master/precio_gasolina.csv'

# Creacion Dataframe
df = pd.read_csv(url, sep=';', decimal=',')

# Visualización del dataframe (la cabecera)
df.head()

Provincia  ... Fecha_extraccion
0  SANTA CRUZ DE TENERIFE  ...       01 08 2020
1  SANTA CRUZ DE TENERIFE  ...       01 08 2020
2  SANTA CRUZ DE TENERIFE  ...       01 08 2020
3  SANTA CRUZ DE TENERIFE  ...       01 08 2020
4  SANTA CRUZ DE TENERIFE  ...       01 08 2020

[5 rows x 13 columns]

**ANALISIS DE TIPO DE VARIABLES, CLASIFICACIÓN DESDE EL PUNTO DE VISTA DEL ANALISIS DE DATOS**


In [5]:
# Resumen información del fichero
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9991 entries, 0 to 9990
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Provincia         9991 non-null   object 
 1   Localidad         9991 non-null   object 
 2   Codigo_postal     9991 non-null   int64  
 3   Direccion         9991 non-null   object 
 4   Margen            9991 non-null   object 
 5   Toma_de_datos     9991 non-null   object 
 6   Precio            9991 non-null   float64
 7   Rotulo            9990 non-null   object 
 8   Tipo_venta        9991 non-null   object 
 9   Rem               9991 non-null   object 
 10  Horario           9991 non-null   object 
 11  Tipo_servicio     9845 non-null   object 
 12  Fecha_extraccion  9991 non-null   object 
dtypes: float64(1), int64(1), object(11)
memory usage: 1014.8+ KB
None


In [6]:
print("Existen un total de " + str(len(df)) + " registros")

Existen un total de 9991 registros


**- Variable: Provincia**

Es la identificación de la Provincia donde se encuentra la Estación de Servicio (Gasolinera).

In [7]:
df.Provincia.unique()

array(['SANTA CRUZ DE TENERIFE', 'MELILLA', 'PALMAS LAS ', 'CEUTA',
       'BADAJOZ', 'SALAMANCA', 'VALENCIA', 'BARCELONA', 'ALMERIA', 'JAEN',
       'GIRONA', 'LLEIDA', 'TARRAGONA', 'ZARAGOZA', 'MURCIA', 'CORDOBA',
       'HUESCA', 'CASTELLON  CASTELLO', 'SEVILLA', 'GUADALAJARA',
       'CANTABRIA', 'CIUDAD REAL', 'CUENCA', 'CADIZ', 'GRANADA', 'TOLEDO',
       'VALLADOLID', 'TERUEL', 'ALBACETE', 'NAVARRA', 'MALAGA',
       'ALICANTE', 'HUELVA', 'MADRID', 'CORUNA A ', 'RIOJA LA ',
       'PONTEVEDRA', 'BURGOS', 'ZAMORA', 'ASTURIAS', 'LUGO', 'CACERES',
       'BIZKAIA', 'SORIA', 'LEON', 'ARABA ALAVA', 'GIPUZKOA', 'SEGOVIA',
       'PALENCIA', 'AVILA', 'OURENSE', 'BALEARS ILLES '], dtype=object)

In [8]:
len(df.Provincia.unique())

52

In [9]:
print(df.Provincia.dtypes)

object


Podemos ver que existen 52 valores diferentes dentro de la data suministrada. Así podemos identificarla como una variable **cualitativa nominal** ya que son categorias sin un orden. Se debe considerar para el análisis de datos 
como **variables categoricas o de clasificación**.

**- Variable: Localidad**

Es la identificación de la ciudad o localidad de la Estación de Servicio.

In [10]:
df.Localidad.unique()

array(['CUESTA LA        ', 'LA LAGUNA', 'SANTA CRUZ DE TENERIFE  ', ...,
       'MONTSENY         ', 'VERGER EL        ', 'BAUL           '],
      dtype=object)

In [11]:
len(df.Localidad.unique())

3887

Al igual que la variable anterior esta se puede identificar como una **variable cualitativa del tipo nominal**, donde existen 3887 clasificaciones por lo que es una variable categorica o de clasificación.

**- Variable: Codigo_postal**

Es un valor numerico único que identifica un codigo asociado a una dirección especifíca o a una localidad especifíca.

In [12]:
df.Codigo_postal.unique()

array([38320, 38200, 38110, ...,  6228,  3770, 18860])

In [13]:
df.Codigo_postal.value_counts()

41500    23
11500    22
4700     21
11130    21
30500    20
         ..
36612     1
36670     1
36890     1
33570     1
18860     1
Name: Codigo_postal, Length: 4198, dtype: int64

Esta variable al ser numerica en su naturaleza se considera como **cuantitativa** y se puede considerar como **continua**, ya que no tiene sentido realizar operaciones aritmeticas sobre ellas. Desde un punto de vista analísitco se considera como **Categorica o de Clasificación**, ya que existen más de una Estación de Servicio asociada a un Código Postal, lo cual permite agrupar por esta variable, como se visualiza en el value_counts().

**- Variable: Direccion**

Identifica a la dirección de la Estación de Servicio, esta debería ser unica para cada uno de las Estaciones.

In [14]:
len(df.Direccion.unique())

9699

In [15]:
df.Direccion.value_counts()

AVENIDA ANDALUCIA SN               11
AVENIDA JUAN CARLOS I SN            7
AVENIDA CONSTITUCION SN             5
AVENIDA DEL MEDITERRANEO SN         4
AVENIDA REINA SOFIA SN              3
                                   ..
AVENIDA FEDERICO SILVA MUNOZ 21     1
CR N631 43 4                        1
CARRETERA N620 KM 307 000           1
CARRETERA TOROSALAMANCA KM 21       1
AUTOVIA A92 KM 322                  1
Name: Direccion, Length: 9699, dtype: int64

Vemos que existen repeticiones en algunas direcciones de Estación de Servicios, por lo que se debe considerar como una variable del tipo **cualitativa nominal**, y que representa una **clasificación o categoria**.

**- Variable: Margen**

El margen indica si cuando vas conduciendo la Estación de Servicio está a la derecha o a la izquierda. Debería tomar dos valores representada por D y N.

In [16]:
df.Margen.unique()

array(['D', 'N', 'I'], dtype=object)

Podemos ver que existen 3 valores, ya que N es aplicable para valores donde NO Aplica esta condición.

In [17]:
df.Margen.value_counts()

D    5016
I    2596
N    2379
Name: Margen, dtype: int64

Considerando que los tres valores tienen una gran cantidad de registros se debe considerar como variable de **Clasificación** del tipo Cualitativa nominal, aunque debería ser interpetrada como del tipo Cualitativa binaria simetrica ya que siempre se puede distinguir si esta a la Izquierda o Derecha.



**- Variable: Toma_de_datos**

Se refiere a la fecha y hora asociada a la toma de dato relacionada al precio.

En este caso, aunque la variable se consume como un string, esta representa un valor numerico, por lo que se considera como una variable del **tipo Cuantitativa continua** y es clasificada como una variable analitica o cuantitativa. 

**- Variable: Precio**

Se refiere al valor de la Gasolina de 95 octanos en el periodo indicado en toma de datos, la cual esta representada por un valor numerico que incluye decimales.

Esta variable es del tipo **Cuantitativa continua** y clasificada como **analitica**, ya que es la variable de medición del dataset.

**- Variable: Rotulo**

Esta variable representa a la Empresa o distribuidor de Gasolina asociada a la Estación de Servicio.

In [18]:
df.Rotulo.unique()

array(['PCAN', 'OCEANO', 'GMOIL', ..., 'HELLIN GAS S L ',
       'MURILLO ROJO SL', 'BIOIL'], dtype=object)

In [19]:
df.Rotulo.value_counts()

REPSOL                            2698
CEPSA                             1330
GALP                               508
SHELL                              365
BP                                 211
                                  ... 
ESTACION DE SERVICIO ARALAR          1
E S KONTXA S L                       1
GASOILS CARRERAS DE PIERA S L        1
TAXCO                                1
BIOIL                                1
Name: Rotulo, Length: 2850, dtype: int64

Esta variable  **Categorica** del tipo **Cualitativa nominal**, ya que existen grandes consorcios que tienen asociados a más de una Estación de Servicios, por lo que se pueden clasificar por esta variable. 

**- Variable: Tipo Venta**

Representa a la forma de venta asociada al precio.

In [20]:
df.Tipo_venta.unique()

array(['P', 'A', 'R', 'N', 'p'], dtype=object)

Esta varible en su naturaleza es **Cualitativa nominal**, ya que representa a más de una forma de venta y no tiene un orden establecido. Asi se puede clasificar como una variable de **Clasificación o Categorica**, ya que podemos agrupar por sus valores.

**- Variable: Rem**

Se refiere al origen de los datos procedentes, indicando 2 valores.

In [21]:
df.Rem.unique()

array(['dm', 'OM'], dtype=object)

Esta variable al tener dos valores unicos se puede considerar en su naturaleza como una variable **Cualitativa Binaria** y desde el punto de analisis de datos es de **Clasificación o Categorica**.

**- Variable: Horario**

Variable que representa el horario de atención de la Estación de Servicio.

In [22]:
df.Horario

0                             LD 24H
1             LS 06002200 D 07002200
2       LV 24H S 06002000 D 08001500
3             LS 07302100 D 08001400
4             LS 07002200 D 09001600
                    ...             
9986           L 07002200 D 07001430
9987                          LD 24H
9988                          LD 24H
9989                      L 09002000
9990                          LD 24H
Name: Horario, Length: 9991, dtype: object

Se puede observar que esta variable toma muchos valores y en diferentes formatos, siendo que en su naturaleza expresa un numero y se considera como **Cuantitativa nominal**, ahora para analisis de datos debemos considerarla como **Categorica**, ya que estas pueden agrupar registros por horarios de atención.